### PR TITLE
Feature/crashes delegate macos

### PR DESCRIPTION
**Pull request by jaelim-ms**
I reimplemented Vendor's PR due to conflicts.
https://github.com/Microsoft/mobile-center-sdk-ios/pull/572

### CODE REVIEW

File: SasquatchMac/SasquatchMacSwift/AppDelegate.swift [hunk 1]
```diff
@@ -23,4 +44,24 @@ class AppDelegate: NSObject, NSApplicationDelegate {
       sasquatchMacView.mobileCenter = MobileCenterDelegateSwift()
     }
   }
+
+  // Crashes Delegate
+  
+  func crashes(_ crashes: MSCrashes!, shouldProcessErrorReport errorReport: MSErrorReport!) -> Bool {
```
Comment (clpolet) [inline SasquatchMac/SasquatchMacSwift/AppDelegate.swift]: We should print something in those delegates, same as the ObjC one.

File: Vendor/macOS/MSAlertController/MSAlertController.m [hunk 2]
```diff
@@ -0,0 +1,39 @@
+#import "MSAlertController.h"
+
+// NSAlertFirstButtonReturn	= 1000
```
Comment (clpolet) [inline Vendor/macOS/MSAlertController/MSAlertController.m]: Maybe remove this line if not needed.

File: Vendor/macOS/MSAlertController/MSAlertController.h [hunk 3]
```diff
@@ -0,0 +1,27 @@
+#import <Cocoa/Cocoa.h>
+#import <Foundation/Foundation.h>
+
+typedef void (^actionCallback)();
+
+NS_ASSUME_NONNULL_BEGIN
+
+@interface MSAlertController : NSAlert
+
++ (MSAlertController *)alertControllerWithTitle:(NSString *)title message:(NSString *)message style:(NSAlertStyle)style;
```
Comment (clpolet) [inline Vendor/macOS/MSAlertController/MSAlertController.h]: Missing documentation on this one.

File: SasquatchMac/SasquatchMacObjC/AppDelegate.m [hunk 4]
```diff
@@ -20,8 +21,79 @@ - (void)applicationDidFinishLaunching:(NSNotification *)aNotification {
...
+  }
+
+  [MSCrashes setDelegate:self];
+  [MSCrashes setUserConfirmationHandler:(^(NSArray<MSErrorReport *> *errorReports) {
+
+               // Use MSAlertViewController to show a dialog to the user where they can choose if they want to provide a
+               // crash
+               // report.
+               MSAlertController *alertController = [MSAlertController
+                   alertControllerWithTitle:@"Sorrty about that!"
```
Comment (clpolet) [inline SasquatchMac/SasquatchMacObjC/AppDelegate.m]: "Sorry" :)

File: SasquatchMac/SasquatchMacObjC/AppDelegate.m [hunk 5]
```diff
@@ -20,8 +21,79 @@ - (void)applicationDidFinishLaunching:(NSNotification *)aNotification {
...
 - (void)setMobileCenterDelegate {
-  SasquatchMacViewController *viewController = (SasquatchMacViewController *) [[[NSApplication sharedApplication] mainWindow] contentViewController];
+  SasquatchMacViewController *viewController =
+      (SasquatchMacViewController *)[[[NSApplication sharedApplication] mainWindow] contentViewController];
   viewController.mobileCenter = [MobileCenterDelegateObjC new];
 }
 
+#pragma mark - Private
+
+- (void)setupCrashes {
```
Comment (clpolet) [inline SasquatchMac/SasquatchMacObjC/AppDelegate.m]: I don't see where you call `setupCrashes`?

File: Vendor/macOS/MSAlertController/MSAlertController.m (hunk 2 above)
Comment (erychagov) [inline Vendor/macOS/MSAlertController/MSAlertController.m]: I thought to use it as description why DEFAULT_SHIFT equals 1000.

Comment (clpolet) [inline Vendor/macOS/MSAlertController/MSAlertController.m]: That's an idea, probably also add some text with it.

**Comment by erychagov** [comment]
https://github.com/Microsoft/mobile-center-sdk-ios/pull/591

**Pull request merged**